### ISSUE TITLE
Sync initial model other than set fixed random seed 

### ISSUE THREAD

**Issue by qshzh** [pytorch/xla#2911]
Thanks for this amazing package.

I see some colab demos, it suggests that using [a fixed random seed](https://github.com/pytorch/xla/blob/ec130ecb003fe3b0a781f4f8b743da2da740982f/test/test_train_mp_mnist.py#L62) to guarantee that the initial models across different process share the same weight.

Can we use another way to sync initial weights? During my training, I need to generate random noise. If I explicitly set the same seed, then every process will generate the same noise, which could do harm to my model training.

In PyTorch DDP, `DistributedDataParallel` can help sync weight at the very beginning. I do not find a similar API in xla. I have a simple implementation, but not sure whether or not there are more elegant implementations
```python
checkpoint_path = osp.join(tempfile.gettempdir(), "initial_weights.pt")
if xm.is_master_ordinal():
    xm.save(model.state_dict(), checkpoint_path)

xm.rendezvous("sync init model")
model.load_state_dict(
    torch.load(checkpoint_path, map_location=device)
)
```

**Comment by jysohn23** [comment]
Yeah the way you're doing it would work on a single host setup, but wouldn't really work across hosts unless you're storing the model weights to some shared filesystem/storage. But we don't currently have something equivalent to DDP which you describe.

Alternatively, you could just set the seed first at the beginning, initialize the model weights and then reset the seed to some unique number like `xm.get_ordinal()` before generating random numbers so that each replica generates different noise?

**Comment by qshzh** [comment]
Thanks for your suggestion. Playing with random seeds is faster.
